feature/visited/src/main/kotlin/uk/govuk/app/visited/data/model/VisitedItem.kt

```diff
@@ -0,0 +1,12 @@
+package uk.govuk.app.visited.data.model
+
+import io.realm.kotlin.types.RealmObject
+import io.realm.kotlin.types.annotations.PrimaryKey
+import org.mongodb.kbson.ObjectId
+
+class VisitedItem: RealmObject {
+    @PrimaryKey var _id: ObjectId = ObjectId()
```

Comment:
Is an id required, can we not just use the url as the primary key?

Reply:
This is an interesting question!

Basically, we could use the URL - but they are not unique over time (like most 'business' keys) and that isn't going to help the user or us at all, and it won't stop multiple entries pointing to the same page. So, while they look like a good option, they really aren't.

Back story... 

We're using (effectively) a content item. These have a title, a content ID, a document type and a path. The path is built from the title, plus the document type - both of which can be changed by the user. This means the path can also change (triggered by the user). They have 3 options: leave the path as-is, update it based on the new title and/or document type, or redirect the old page to the new one. If we were using the content ID - which is unique and unchanging for the entire lifecycle of the content item - then that would be a clear preference for the key here.

So, my thinking is - if we can't find a useful 'business' key - then we shouldn't use one as it'll send the wrong message to other devs on the app.

Reply:
So this means we could get multiple entries for the same url but some of them will have title's that don't match what is currently on the page?

Reply:
Yes, it is possible.

Reply:
What would happen if you access the same page multiple times (not from the visited pages screen)? I'm thinking you would get multiple entries, is this the desired behaviour?

Reply:
That wouldn't be desirable no. However, when we `insertOrUpdate` we check for existing visited items by title and url - should stop this happening I think.

Reply:
So, we don't use the id for this reason.